lib/names/normalize.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package names
+
+import (
+	"regexp"
+	"strings"
+)
+
+var (
+	nonNameChar               = regexp.MustCompile("[^-.a-z0-9]+")
+	dotDashSeq                = regexp.MustCompile("[.-]*[.][.-]*")
+	trailingLeadingDotsDashes = regexp.MustCompile("^[.-]*(.*?)[.-]*$")
+)
+
+// Convert a name to normalized form.  This is used for conversion of os.Hostname to a
+// suitable node name, and is also used for the v2->v3 migration code.
```

Comment:
I think this isn't really what we want to do - users need to pass in a valid hostname, and valid hostnames don't contain `/` characters. We shouldn't try to magically convert them away, we should return an error.